rigging/rigs/packet.py

```diff
@@ -0,0 +1,358 @@
+                getattr(ICMP_TYPES, icmptype_str.upper().replace('-', '_'), None)
+
+        
+        # Fail if the network interface doesn't exist on the system.
+        ifname = self.get_option('ifname')
+        if ifname:
+            try:
+                if_nametoindex(ifname)
+                self._match_ifname = ifname
+            except OSError:
+                raise CannotConfigureRigError(f"Interface \"{ifname}\" doesn't exist.")
+
+        # Build a comparable object if a payload regex is provided
+        payload = self.get_option('payload')
+        if payload:
+            self._must_match['payload'] = PayloadMatch(payload)
+
+        # Remove all attributes that have no value
+        self._must_match = { k: v for k, v in self._must_match.items() if v }
+
+        # No filters provided
+        if not self._must_match:
+            raise CannotConfigureRigError("Must specify at least one filter")
+
+        self.add_watcher_thread(target=self._read_from_socket, args=())
+
+
+    @staticmethod
+    def _strmac(mac):
+        return ":".join([ f"{i:02x}" for i in mac])
+
+    def _pkt_matches(self, pkt_attrs):
+
+        matching_keys = {}
+
+        for k, v in self._must_match.items():
+
+            if not k in pkt_attrs:
+                continue
+
+            # Match tcpflags if any of the provided flags are set
+            if isinstance(v, TCP_FLAGS) and v & pkt_attrs[k] != 0:
+                matching_keys[k] = v
+
+            # For all other keys we just compare them
+            elif pkt_attrs[k] == v:
+                matching_keys[k] = v
+
+
+        if self.get_option("any"):
+            if len(matching_keys) > 0:
+                return matching_keys
+            else:
+                return None
+
+        else:
+            # If --any wasn't set, then all provided filters must match.
+            if len(matching_keys) == len(self._must_match):
+                return matching_keys
+            else:
+                return None
+
+    def _read_from_socket(self):
+
+        sock = socket(AF_PACKET, SOCK_RAW, htons(ETH_P_ALL))
+
+        while True:
+            eth, addrinfo = sock.recvfrom(65535)
+
+            pkt_str = ""
+            pkt_attrs = {}
+            payload = None
+
+            iface, ethtype_info, _, _, srcmac_info = addrinfo
+
+            if self._match_ifname and self._match_ifname != iface:
+                continue
+
+            if ethtype_info != ETH_IPV4: # Only IPv4
+                continue
+
+            # L2
+
+            eth_src = self._strmac(eth[:6])
+            eth_dst = self._strmac(eth[6:12])
+            #eth_type = struct.unpack("!H", eth[12:14])[0]
+
+            pkt_attrs["srcmac"] = eth_src
+            pkt_attrs["dstmac"] = eth_dst
+
+            # L3
+
+            ip = eth[14:]
+
+            ip_ver, ip_hdrlen = divmod(ip[0], 16)
+            ip_hdrlen *= 4 # 32bit incrs
+
+            if ip_ver != 4:
+                continue
+
+            ip_diffserv = ip[1]
```

Comment:
LGTM highlights this as unused.

Is there a use case for it? If so, let's flesh that out, otherwise drop it.

Reply:
I don't think there's a use case. We can add it later on if needed. Removing it for now